### PR TITLE
CLOUDP-175229: fix k8s resource mapping

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes
fix k8s resource mapping

_Jira ticket:_ CLOUDP-175229

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/kubernetes/operator/features/crds.go [hunk 1]
```diff
@@ -71,9 +71,18 @@ var (
...
 )
 
+func GetResourcesForVersion(version string) ([]string, bool) {
+	semVer := strings.Split(version, ".")
+	semVer[2] = "0"
+	majorVersion := strings.Join(semVer, ".")
+
+	resources, ok := versionsToResourcesMap[majorVersion]
+	return resources, ok
+}
```
Comment (gssbzn) [inline internal/kubernetes/operator/features/crds.go:81]: we have https://github.com/Masterminds/semver, would it help here?

**Review by gssbzn — APPROVED** [review]
LGTM if LGTEVG

**Comment by helderjs** [comment]
@tibulca wouldn't this PR be a duplicate of this one https://github.com/mongodb/mongodb-atlas-cli/pull/1889/files
It was responsible for breaking the test and the fix was also introduced there

**Comment by helderjs** [comment]
Ouch, seems I need to have a better understanding of the evg status

File: internal/kubernetes/operator/features/crds.go (hunk 1 above)
Comment (tibulca) [reply internal/kubernetes/operator/features/crds.go]: yes, we can use it. I made the changes

**Comment by tibulca** [comment]
> @tibulca wouldn't this PR be a duplicate of this one https://github.com/mongodb/mongodb-atlas-cli/pull/1889/files? It was responsible for breaking the test and the fix was also introduced there

@helderjs that PR didn't fix the test, see [here](https://evergreen.mongodb.com/task/mongodb_atlas_cli_master_e2e_atlas_kubernetes_atlas_kubernetes_e2e_87397580365d8695a64d2655b727081bfe299632_23_04_18_09_38_04)

**Review by blva — APPROVED** [review]

**Comment by helderjs** [comment]
@tibulca @gssbzn Just for clarification. I think I understood what happened here, follow the list the events:
* We released 1.7.2
* When we updated atlascli to support it, it was broken but fixed in the same PR (https://github.com/mongodb/mongodb-atlas-cli/pull/1889/files)
* Seems the Atlas API client  or the command got an update and an object changed behavior, breaking another part of the test. I see you have also added https://github.com/mongodb/mongodb-atlas-cli/pull/1929/files#diff-e14b5c98d99004f30ce73a423412c8b806479e057c4af4ea2d2ba7106d26e036R424. I have also added this in my recent PR(not merged yet)
* Once the ticket was generated by the first error, seems to handle the operator version was the issue, but the real problem is the second one.
* If you look at events after April 25th (when 1.7.2 was merged), the error is other, the second one I mentioned. (we also have timeouts due to cloud-qa being broken)

btw, the PR looks good, I'm just trying to understand watch happened and what I missed

**Pull request merged**